org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java

```diff
@@ -651,6 +657,48 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
 			format(ent);
 	}
 
+	/**
+	 * Format the diff entries by filtering out the noise from the given delta filter pattern
+	 * The filter acts only for files that have MODIFY change Type.
+	 * If there are no changes detected, we will remove the diff entry.
+	 * @param entries
+	 * @param deltaFilterPattern
+	 * @throws IOException
+	 */
+
+	public void format(List<? extends DiffEntry> entries, Pattern deltaFilterPattern) throws IOException {
+		if (deltaFilterPattern == null)
+			format(entries);
+		else {
+
+			Iterator<? extends DiffEntry> diIterator = entries.iterator();
+
+			while (diIterator.hasNext()) {
+
+				DiffEntry diffEntry = diIterator.next();
+
+				if (MODIFY.equals(diffEntry.changeType)) {
+
+					FormatResult res = createFormatResult(diffEntry);
+
+					String aContent = new String(res.a.content);
+					String bContent = new String(res.b.content);
+
+					aContent = deltaFilterPattern.matcher(aContent).replaceAll(EMPTY_STRING);
+					bContent = deltaFilterPattern.matcher(bContent).replaceAll(EMPTY_STRING);
+
+					if (!aContent.equals(bContent)) {
+						format(res.header, res.a, res.b);
+					} else {
+						diIterator.remove();
+					}
```

Comment:
Break this into a private helper method.

Reply:
ah, I can do it. But its just 6 lines and it's already calling createFormatResult and  format methods.

Reply:
I get it, just makes it easier to read, makes your higher level function more clear in its actual intent.

Reply:
leaving in the same method